### PR TITLE
changes for archive-it

### PR DESCRIPTION
**Pull request by nlevitt**
- replication
- fuzzy canon using pywb rules.yaml
- more logging
- upgrade rocksdb
- command line options for tweaking performance
- filter plugins
- support new query params
  * `urlkey` - alternative to `url`, bypasses outbackcdx canonicalization
  * `from` - minimum timestamp
  * `to` - maximum timestamp
  * `collapse` aka `collapseToFirst`
  * `collapseToLast`

### CODE REVIEW

File: src/outbackcdx/NanoHTTPD.java [hunk 1]
```diff
@@ -1004,4 +1039,80 @@ public final InputStream getInputStream() {
             return bodyStream;
         }
     }
-}
\ No newline at end of file
+
+    // copied from guava
```
Comment (ato) [inline src/outbackcdx/NanoHTTPD.java]: We need to include guava's copyright notice and probably also the author tags from the javadoc (which would seem likely to count as an 'attribution notice'):

> You must retain, in the Source form of any Derivative Works that You distribute, all copyright, patent, trademark, and attribution notices from the Source form of the Work, excluding those notices that do not pertain to any part of the Derivative Works;

Comment (nlevitt) [reply src/outbackcdx/NanoHTTPD.java]: How does this look?
https://github.com/nla/outbackcdx/pull/75/commits/9d29e742fe3e3e9d20a7170f8ecf8da73495f393

Comment (ato) [reply src/outbackcdx/NanoHTTPD.java]: Perfect

**Comment by nlevitt** [comment]
Hey Alex, I think this PR is ready. We're going live with these changes later this week. It'd be cool to run off of master, but no pressure, we can move ahead running off of our branch for now. There are a few little bits of polish that are still missing from this changeset, for example listing all the new options in /config.json, and I'm not sure what else exactly. Not sure if you feel that kind of thing is a blocker for merge.

**Comment by ato** [comment]
Awesome! Stoked to be merging this. :-)
I've released it as 0.8.0.

**Comment by nlevitt** [comment]
Thank you @ato by the way! (For the record, the new release is 0.7.0)

**Pull request merged**